include/aie/Dialect/AIEX/IR/AIEX.td

```diff
@@ -463,9 +464,29 @@ def AIE_SelectOp: AIEX_Op<"select", []>, Results<(outs Index)> {
   ];
 }
 
+def AIE_RuntimeSequenceOp : AIEX_Op<"runtime_sequence", [NoTerminator, HasParent<"AIE::DeviceOp">]> {
+  let summary = "Program the configuration co-processor of the AI Engine array";
+  let description = [{
+    Instructions in this operation allow for runtime (re-)configuration of the AI Engine array, such as configuring data movement buffer descriptors.
+    These instructions will execute on the configuration co-processor of the AI Engine array.
+
+    Typically, these instructions include configuring the data transfers between host and AIE array on the shims.
+    The input arguments are arguments passed in from the host at kernel invocation time. This may include buffers on the host.
+  }];
+  let arguments = (ins
+    Variadic<AnyType>:$args
+  );
+  let regions = (region
+    AnyRegion:$body
+  );
+  let hasCustomAssemblyFormat = 1;
```

Comment:
I'm not sure you need a custom parser and printer for this op.
Does something like this work?
```
'(' ($args^ `:` type($args))? ')' attr-dict `:` regions
```
I probably don't have the syntax quite right, but something like that should work.

Reply:
Three problems I ran into with that are:
 1. In the function declaration, `%ssa` actually declares a new SSA value. But if you do it the way you suggest, it needs to reference an already-defined value. The whole idea though is that it's supposed to be a new one that you can then use in the function body.
 2.  I wouldn't know how to let the region parser know that it can use those values as entry arguments.
 3. I think `aiex.sequence (%a, %b, %c) : (memref<>, memref<>, memref<>)` is ugly compared to `aiex.sequence (%a: memref<>, %b: memref<>, %c: memref<>)`

If you know how to address 1 and 2 I'm happy to live with 3 though.